sqlmesh/core/context.py

```diff
@@ -228,6 +233,7 @@ class ExecutionContext(BaseContext):
         engine_adapter: The engine adapter to execute queries against.
         snapshots: All upstream snapshots (by model name) to use for expansion and mapping of physical locations.
         deployability_index: Determines snapshots that are deployable in the context of this evaluation.
+        engine_adapters: Optional dictionary of multiple engine adapters.
```

Comment:
Note: discussed about implications of this offline. We can't just deprecate `engine_adapter` in `ExecutionContext`, because it's used in python models to do dynamic db calls. Same holds for the macro evaluator's `engine_adapter` attribute, which needs to match the corresponding model's gateway.

Reply:
you need the property but you don’t need to pass it in right

Reply:
Why? How will the execution context know which adapter to use then for a python model that specifies a different gateway than the default?